library/src/main/java/com/chuckerteam/chucker/internal/support/FormatUtils.kt

```diff
@@ -0,0 +1,178 @@
+package com.chuckerteam.chucker.internal.support
+
+import android.content.Context
+import android.text.TextUtils
+import com.chuckerteam.chucker.R
+import com.chuckerteam.chucker.internal.data.entity.HttpHeader
+import com.chuckerteam.chucker.internal.data.entity.HttpTransaction
+import com.google.gson.JsonParseException
+import com.google.gson.JsonParser
+import java.io.ByteArrayInputStream
+import java.io.ByteArrayOutputStream
+import java.io.PrintWriter
+import java.io.StringWriter
+import java.util.Locale
+import javax.xml.XMLConstants
+import javax.xml.transform.OutputKeys
+import javax.xml.transform.TransformerException
+import javax.xml.transform.sax.SAXSource
+import javax.xml.transform.sax.SAXTransformerFactory
+import javax.xml.transform.stream.StreamResult
+import kotlin.math.ln
+import kotlin.math.pow
+import org.xml.sax.InputSource
+
+internal object FormatUtils {
+
+    private const val SI_MULTIPLE = 1000
+    private const val BASE_TWO_MULTIPLE = 1024
+
+    fun formatHeaders(httpHeaders: List<HttpHeader>?, withMarkup: Boolean): String {
+        return httpHeaders?.joinToString(separator = "") { header ->
```

Comment:
A bit hard to read (to be fair, the java version as well), could you simply do a if else or when on markup and declare two raw string blocks?